pkg/certs/cert-inspection/certgraphanalysis/nameguessing.go

```diff
@@ -55,22 +56,59 @@ var secretLocationToLogicalName = map[certgraphapi.InClusterSecretLocation]logic
 	newSecretLocation("", ""): newMeaning("", ""),
 }
 
-func guessMeaningForCertKeyPair(in certgraphapi.CertKeyPair) logicalMeaning {
-	for _, loc := range in.Spec.SecretLocations {
-		if meaning, ok := secretLocationToLogicalName[loc]; ok {
-			return meaning
-		}
+func formatSecretLocation(loc certgraphapi.InClusterSecretLocation, nodes map[string]int) certgraphapi.InClusterSecretLocation {
```

Comment:
I'm ok with this approach as your starting point (won't block merge), but I suspect you'll actually want to scan everything and replace the nodename in all secret names, all namespaces, all dns names, all subjects, etc.